### PR TITLE
transformers: 4.33.2 -> 4.34.0

### LINKED ISSUE #259167: Update request: transformers 4.33.2 → 4.34.0
- Package name: python310Packages.transformers
- Latest released version: 4.34.0

- Current version on the unstable channel: 4.33.2
- Current version on the stable/release channel:

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@pashashocky @happysalada

-----

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by suhr**
## Description of changes

Closes #259167.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
the names of the commits need to be
`python311Packages.tokenizers: 0.13.3 -> 0.14.0`
and same for the transformers, otherwise ofborg doesn't build the new packages.

File: pkgs/development/python-modules/tokenizers/default.nix [hunk 1]
```diff
@@ -60,21 +60,18 @@ let
...
     owner = "huggingface";
     repo = pname;
-    rev = "python-v${version}";
-    hash = "sha256-QZG5jmr3vbyQs4mVBjwVDR31O66dUM+p39R0htJ1umk=";
+    rev = "v${version}";
+    hash = "sha256-zCpKNMzIdQ0lLWdn4cENtBEMTA7+fg+N6wQGvio9llE=";
   };
 
-  postPatch = ''
-    ln -s ${./Cargo.lock} Cargo.lock
```
Comment (happysalada) [inline pkgs/development/python-modules/tokenizers/default.nix:75]: if we don't need then cargo.lock anymore, should we remove the file completely from the filesystem ?

Comment (suhr) [reply pkgs/development/python-modules/tokenizers/default.nix:75]: importCargoLock uses it:

```
  cargoDeps = rustPlatform.importCargoLock {
    lockFile = ./Cargo.lock;
  };
```

Comment (happysalada) [reply pkgs/development/python-modules/tokenizers/default.nix:75]: Missed this, thank you.
If you can just change the commit descriptions then, we wait for ofborg confirmation and this should be good to go.

**Comment by happysalada** [comment]
Thank you for this!
I don't know if you know, but it should take a couple of days for the unstable channel to be updated. If you need to use this straight away, you can always use master.

**Pull request merged**